packages/@aws-cdk/aws-appsync/lib/data-source.ts

```diff
@@ -203,6 +203,35 @@ export class DynamoDbDataSource extends BackedDataSource {
   }
 }
 
+/**
+ * The authorization config in case the HTTP endpoint requires authorization
+ */
+export interface HttpDataSourceAuthorizationConfig {
+  /**
+   * The authorization type required by the HTTP endpoint
+   */
+  readonly authorizationType: 'AWS_IAM';
+  /**
+   * The IAM configuration required by the HTTP endpoint
+   */
+  readonly awsIamConfig: HttpDataSourceIamConfig;
+}
+
+/**
+ * The IAM configuration required by the HTTP endpoint
+ */
+export interface HttpDataSourceIamConfig {
+  /**
+   * The signing region for AWS IAM authorization
+   */
+  readonly signingRegion: string;
+
+  /**
+   * The signing service name for AWS IAM authorization
+   */
+  readonly signingServiceName: string;
+}
```

Comment:
If the authorization type is always going to AWS_IAM as seen by these [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appsync-datasource-authorizationconfig.html), I think it might be best to just expose `awsIamConfig` and remove that additional level of abstraction!

Reply:
@BryanPan342 I have a question.

JSII does not allow nested objects directly. I got a linting error when I tried to implement like below.

`Only string-indexed map types are supported`

```ts
export interface AwsIamConfig {
  /**
   * The authorization type required by the HTTP endpoint
   */
  readonly authorizationType: 'AWS_IAM';
  /**
   * The IAM configuration required by the HTTP endpoint
   */
  readonly awsIamConfig: {
    /**
     * The signing region for AWS IAM authorization
     */
    readonly signingRegion: string;

    /**
     * The signing service name for AWS IAM authorization
     */
    readonly signingServiceName: string;
  }
}
`

Do you know the better way to integrate it?